x-pack/plugins/fleet/server/saved_objects/migrations/security_solution/to_v7_15_0.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectMigrationFn, SavedObjectUnsanitizedDoc } from 'kibana/server';
+import { cloneDeep } from 'lodash';
+
+import type { PackagePolicy } from '../../../../common';
+
+export const migratePackagePolicyToV7150: SavedObjectMigrationFn<PackagePolicy, PackagePolicy> = (
+  packagePolicyDoc
+) => {
+  const updatedPackagePolicyDoc: SavedObjectUnsanitizedDoc<PackagePolicy> = cloneDeep(
+    packagePolicyDoc
+  );
+  if (packagePolicyDoc.attributes.package?.name === 'endpoint') {
```

Comment:
Minor efficiency suggestion:

check for Endpoint package name as soon as the first thing in this, and before doing a `cloneDeep()`. If not a endpoint package entry, then just return the `packagePolicyDoc` from the input arguments.